core/src/main/java/org/apache/accumulo/core/clientImpl/bulk/BulkImport.java

```diff
@@ -616,7 +616,9 @@ private Text toText(byte[] row) {
         if (containsPrevRow && containsEndRow) {
           mappings.get(ke).merge(mappings.remove(oke));
         } else {
-          throw new RuntimeException("TODO handle merges");
+          throw new RuntimeException("Unable to execute merge. Key extent "
+                  + oke.toString() + " extends beyond the range of Key extent "
+                  + ke.toString() + ". Out of range data would be lost.");
```

Comment:
This wording may be confusing to users.  Even though we are merging keys, it is happening mostly behind the scenes during a bulk import.  Perhaps something like:
```suggestion
          throw new RuntimeException("Error during bulk import: Unable to merge overlapping keys during a table split.  Key extent " + oke.toString() + " extends beyond the range of " + ke.toString());
```
@keith-turner does this message make sense in the context of this method?

Reply:
I will look at this next week.

Reply:
I looked into this a bit and I don't think the existing code is correct.  I think it should do something like :

```java
        if (ke.contains(oke)) {
           mappings.get(ke).merge(mappings.remove(oke));
        } else if (!oke.contains(ke)) {
         throw new RuntimeException("Error during bulk import: Unable to merge overlapping tablets where neither tablet contains the other.  This may be caused by a concurrent merge.  Key extents " + oke + " and " + ke+" overlap, but neither contains the other"));
        }
```

The above code assumes the code in this method to do the overlapping check is moved to KeyExtent.contains(KetExtent) to make the check simpler.

Reply:
@keith-turner I agree this is better than the original. We can remove the 2 booleans from lines 611-614, making things more succinct. I will update the PR.

Reply:
@keith-turner Looks like contains() needs a ByteSequence or a BinaryComparable as the parameter. I will look into trying to convert the KeyExtents to one of these datatypes.

Reply:
@cradal  I was thinking this following code could be moved into a KeyExtent.contains(KeyExtent) method.  Maybe something like the following.

```java
boolean contains(KeyExtent oke){
        boolean containsPrevRow = getPrevEndRow() == null || (oke.getPrevEndRow() != null
            && getPrevEndRow().compareTo(oke.getPrevEndRow()) <= 0);
        boolean containsEndRow = getEndRow() == null
            || (oke.getEndRow() != null && getEndRow().compareTo(oke.getEndRow()) >= 0);

       return containsPrevRow && containsEndRow;
}
```

Reply:
@keith-turner I added contains(KeyExtent oke) to the KeyExtent class and updated mergeOverlapping() in BulkImport accordingly.

Reply:
looks good @cradal